lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -87,45 +88,48 @@ public void OnNext(IDriverStarted value)
 
             public void OnNext(IAllocatedEvaluator value)
```

Comment:
We need to check whether AppVeyor passes without `ICompletedTask` handler (because not having a handler for any event should be acceptable for any REEF driver), but for functionality purposes we still need to check for task failure here (otherwise we're back to where we started - not knowing whether task fails if evaluator fails).

I'm fine with the move of poisoning to `IActiveContext` as we have now.